### PR TITLE
Eperez generated passwd

### PR DESCRIPTION
**Pull request by enriquepablo**
This goes on top of #152

Add a boolean property to password credentials to store whether the password was generated (True) or custom (False)

### CODE REVIEW

File: src/eduid_userdb/credentials/password.py [hunk 1]
```diff
@@ -107,25 +111,37 @@ def credential_id(self, value):
...
     @salt.setter
-    def salt(self, value):
+    def salt(self, value: str):
         """
         :param value: Password salt.
-        :type value: str
         """
         if not isinstance(value, string_types):
-            raise UserDBValueError("Invalid 'salt': {!r}".format(value))
+            raise UserDBValueError("Invalid 'salt': {value}")
```
Comment (johanlundberg) [inline src/eduid_userdb/credentials/password.py]: Missing f before the string, right?

File: src/eduid_userdb/reset_password/db.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        :raise self.DocumentDoesNotExist: No document match the search criteria
+        :raise self.MultipleDocumentsReturned: More than one document matches
+                                               the search criteria
+        """
+        state = self._get_document_by_attr('eduPersonPrincipalName', eppn,
+                                           raise_on_missing)
+        if state:
+            return self.init_state(state)
+
+    @staticmethod
```
Comment (johanlundberg) [inline src/eduid_userdb/reset_password/db.py]: Would it not be better with at classmethod?

File: src/eduid_userdb/credentials/password.py [hunk 3]
```diff
@@ -35,7 +35,7 @@
 from __future__ import absolute_import
 
 import copy
-
+from typing import Optional, Union
 from bson.objectid import ObjectId
 from six import string_types
```
Comment (johanlundberg) [inline src/eduid_userdb/credentials/password.py]: six can be removed in the files we touch now

File: src/eduid_userdb/deprecation.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+import warnings
+import inspect
+from six import string_types
```
Comment (johanlundberg) [inline src/eduid_userdb/deprecation.py]: six can be removed in the files we touch now

File: src/eduid_userdb/reset_password/state.py [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+import bson
+import copy
+import datetime
+from typing import cast, Union, Optional, Dict, Mapping
+from six import string_types
```
Comment (johanlundberg) [inline src/eduid_userdb/reset_password/state.py]: six can be removed in the files we touch now

File: src/eduid_userdb/deprecation.py (hunk 4 above)
Comment (johanlundberg) [reply src/eduid_userdb/deprecation.py]: But if this code is copied verbatim from stackoverflow it should maybe stay as is.

File: src/eduid_userdb/credentials/password.py (hunk 1 above)
Comment (enriquepablo) [reply src/eduid_userdb/credentials/password.py]: Oops right

File: src/eduid_userdb/reset_password/db.py (hunk 2 above)
Comment (enriquepablo) [reply src/eduid_userdb/reset_password/db.py]: Not sure what you mean - To use the provided cls to instantiate the `resetpw_state`? Then we'd be overriding whatever comes in `state.get('method')`, is that right?

**Comment by enriquepablo** [comment]
I'll close and open again, this is showing changes from a previous PR already merged.

**Comment by enriquepablo** [comment]
reopen

**Comment by enriquepablo** [comment]
still showing changes from a previous PR already merged.

**Pull request closed without merging**